crates/oxc_language_server/src/main.rs

```diff
@@ -180,8 +188,38 @@ impl LanguageServer for Backend {
         }
     }
 
-    async fn did_change_watched_files(&self, _params: DidChangeWatchedFilesParams) {
+    async fn did_change_watched_files(&self, params: DidChangeWatchedFilesParams) {
         debug!("watched file did change");
+        if self.options.lock().await.use_nested_configs {
+            let mut config_files = FxHashMap::default();
+            let existing_nested_configs = (*self.nested_configs.read().await).clone();
+            for (key, value) in existing_nested_configs {
+                config_files.insert(key.clone(), value.clone());
+            }
+
+            params.changes.iter().for_each(|x| {
+                if x.uri.to_file_path().unwrap().file_name().unwrap() != OXC_CONFIG_FILE {
+                    return;
+                }
+                // spellchecker:off -- "typ" is accurate
+                if x.typ == FileChangeType::CREATED || x.typ == FileChangeType::CHANGED {
+                    // spellchecker:on
+                    let file_path = x.uri.to_file_path().unwrap();
```

Comment:
we could avoid a duplicate unwrap here and some work by saving `to_file_path()`:

```suggestion
                let Some(file_path) = x.uri.to_file_path() else {
                    return;
                };
                if file_path.file_name().unwrap() != OXC_CONFIG_FILE {
                    return;
                }
                // spellchecker:off -- "typ" is accurate
                if x.typ == FileChangeType::CREATED || x.typ == FileChangeType::CHANGED {
                    // spellchecker:on
```